Tangem/App/Services/CommonSwappingWalletDataProvider/CommonSwappingWalletDataProvider.swift

```diff
@@ -107,6 +107,25 @@ extension CommonSwappingWalletDataProvider: SwappingWalletDataProvider {
         }
         return balance
     }
+
+    func getAllowance(for currency: Currency, from spender: String) async throws -> Decimal {
+        guard let contractAddress = currency.contractAddress else {
+            throw SwappingManagerError.contractAddressNotFound
+        }
+
+        var allowance = try await ethereumNetworkProvider.getAllowance(
+            owner: walletAddress,
+            spender: spender,
+            contractAddress: contractAddress
+        ).async()
+
+        allowance = currency.convertFromWEI(value: allowance)
+        return allowance
```

Comment:
Так очевиднее. А вообще хотелось бы сразу не в копейках получать

```suggestion
        let allowanceInWEI = try await ethereumNetworkProvider.getAllowance(
            owner: walletAddress,
            spender: spender,
            contractAddress: contractAddress
        ).async()

        return currency.convertFromWEI(value: allowanceInWEI)
```

Reply:
Вообще эту штука надо где то зафиксировать в какой момент все переводить из копеек.
Часто эта штука стреляет

Reply:
Лучше чтобы наружу копейки не торчали, у нас везде decimal. Копейки это скорее более низкоуровневое сдкашное что-то

Reply:
тут просто получается, надо еще decimalValue передавать в метод, что бы знать на что делить
а по факту allowance в копейках считается. Хз, обсудим командой, хочется и удобства и единообразия